R/R/flow.R

```diff
@@ -0,0 +1,102 @@
+                        stopifnot(is.character(name), length(name) == 1)
+                        private$.name <- name
+                        private$.flow_decorators <- flow_decorators
+                      },
+                      format = function() {
+                        print_flow(
+                          flow = private$.name,
+                          flow_decorators = private$.flow_decorators,
+                          parameters = private$.parameters,
+                          steps = private$.steps
+                        )
+                      },
+                      add_parameter = function(x) {
+                        if (!is.null(private$.parameters)) {
+                          private$.parameters <- c(private$.parameters, x)
+                        } else {
+                          private$.parameters <- x
+                        }
+                        invisible(self)
+                      },
+                      add_step = function(x) {
+                        private$.steps <- c(private$.steps, x)
+                        invisible(self)
+                      },
+                      add_function = function(x) {
+                        if (!is.null(private$.functions)) {
+                          private$.functions <- c(private$.functions, x)
+                        } else {
+                          private$.functions <- x
+                        }
+                        invisible(self)
+                      },
+                      get_flow = function(save = FALSE) {
+                        x <- print_flow(
+                          private$.name,
+                          private$.flow_decorators,
+                          private$.parameters,
+                          paste0(private$.steps, collapse = "")
+                        )
+                        if (save) {
+                          writeLines(x, con = "flow.py")
+                        } else {
+                          return(x)
+                        }
+                      },
+                      get_name = function() {
+                        private$.name
+                      },
+                      get_parameters = function() {
+                        private$.parameters
+                      },
+                      get_steps = function() {
+                        private$.steps
+                      },
+                      get_functions = function() {
+                        if (length(private$.functions) == 1) {
+                          private$.functions
+                        } else {
+                          private$.functions[!unlist(lapply(private$.functions, is.null))]
+                        }
+                      }
+                    )
+)
+
+header <- function(flow, flow_decorators = NULL) {
+  imports <- paste0(c("FlowSpec", "step", "Parameter", "retry", "environment", "batch", "timeout", "catch", "resources"), collapse = ", ")
```

Comment:
I am wondering if `metaflow` could expose a method which lists all top-level imports for py3 and we populate this list by making a reticulate call.

Reply:
Are you thinking about something like `--help`? Basically we run `Rscript flow.R --help` and pass `--help` to python so the click library can list relevant command line args? I can give it a try.

Reply:
I've added support for --help. Now users can do `Rscript flow.R --help` and `Rscript flow.R run --help` and see the help messages.